### ISSUE TITLE
[1.18.1] No (default) recipe for ME Bridge?

### ISSUE THREAD

**Issue by grekdawid26** [Seniorendi/AdvancedPeripherals#239]
### Descripe

Hello,
I'm playing on MC 1.18.1 (there was the same issue on 1.18) with some mods (no modpack), significant for this issue would be: JEI, AE2, CC: Tweaked, and the Advanced Peripherals.
Shouldn't there be some default recipe available for ME Bridge? - neither the JEI nor the Minecraft's Crafting Tables' "Green-Book-Thingy" show any recipe for it.

### Steps to reproduce

1. Run the Minecraft
2. Try clicking on the ME Bridge in the JEI, or search for it in Minecraft's Crafting Tables' "Green-Book-Thingy" search box
3. Nothing!

### Multiplayer?

No

### Version

0.7.8r (Latest)

### Minecraft, Forge and maybe other related mods versions

MC 1.18.1, Forge 39.0.7

**Comment by SirEndii** [comment]
Oh, yeah. My fault. Forgot to add a recipe.

**Comment by grekdawid26** [comment]
In the meantime could you specify what that recipe would look like? - _There is no recipe in the docs and I could just drop items into lava and /give myself the ME Bridge - it is a quick solution for me and anybody else wanting to craft it - just /giving it seems like cheating - that's just more complex crafting._

**Comment by SirEndii** [comment]
Sure, give one second

**Comment by SirEndii** [comment]
FIF
IAI
FIF

F = fluix block
I = item interface
A = peripheral casing
This is the recipe

**Comment by SirEndii** [comment]
[advancedperipherals-1.18.1-0.7.8r.zip](https://github.com/Seniorendi/AdvancedPeripherals/files/7757956/advancedperipherals-1.18.1-0.7.8r.zip)
Fixed here
Unzip the file before using

**Comment by grekdawid26** [comment]
Thank you, that's so nice.
But now I'm wondering what was wrong with my brain yesterday - I wanted to point out that AE2 changed their "mod-id" semi-recently - even better I just wanted to check if that's the case and submit a pull request with the corrected one, so I search the recipes folder and was like - "ohhh... there is no ME Bridge recipe, they just forgot to put that in" - and now I saw the commit changing the me_bridge.json file _that was already there_...

**Comment by grekdawid26** [comment]
Btw, should the ME Bridge disappear after breaking?
_I know that should be a separate issue, but it is related and I am not sure if it is an actual issue._

**Comment by grekdawid26** [comment]
I think you missed it here:
- src/generated/resources/data/minecraft/tags/blocks/mineable/pickaxe.json
- src/generated/resources/data/minecraft/tags/blocks/needs_iron_tool.json

I temporarily just added it there in my .jar file, but as there is 'generated' in the path I guess that these files are _generated_ from some other files, so you probably should change it somewhere there, idk how that java stuff works, so yeah...

**Comment by SirEndii** [comment]
Yeah, their are generated with datagenerators. I forgot to run them on the last update.

**Comment by grekdawid26** [comment]
Remember also about that loot table file, it doesn't want to pop out as an item when it is not there ;/
Maybe you can add that to that Gradle stuff (isn't that some java build system?) - if they can run automatically, would make things easier in the future (if possible)

**Comment by grekdawid26** [comment]
Did you intend for ME Bridge to take 500 AE/t of the passive drain (for comparison ME Controller takes 3 AE/t and ME Interface takes 0.5 AE/t) using Vibration Chamber (which generates - at max - 40 AE/t) to generate power - the only way in the AE2, that I know of, to generate it - it would require 13 Vibration Chambers, just to power this one block?
Ignoring the insane amount of Vibration Chambers - at 40 AE/t they burn about 1 coal / 5 sec, so it would take almost 3 coals / s - that is about 10 800 coals per hour to run it.
I am really sorry for so many messages... _but I have found 2 issues..._

**Comment by SirEndii** [comment]
You can change the energy usage in the config
But it should take 500 FE instead of AE(Would be the half)

**Comment by grekdawid26** [comment]
It is still a lot, maybe you set that up when there were some other energy/power related mods - I couldn't find any for 1.18, which means that the only way I can generate power is through the AE's Vibration Chamber.
The 250 AE/t (passive) is a huge amount, event if you would cut that in half to 120 AE/t this is still - a lot considering that it is _passive_ and max amount of energy that Vibration Chamber can generate per tick and its resource consumption.

The other AE2 block/item that I encountered which consumes a similar amount of energy is charging the Wireless stuff, but that is only when you actually perform the action of charging it, so its consumption of energy is momentary and AE2 is responding to that issue by allowing you to buffer the energy in the Energy Cells - additionally the Wireless terminals are using energy only when you use them, so you don't need an insanely HUGE amount of fuel to charge them so often.

Also, the cost-balance is in the Advanced Peripherals seems strange ex. my understanding (from docs) is that the Player Inventory access block does not take any kind of energy and just allow to access player inventory, which is so OP I don't even want to use that thing, but then the ME Bridge which just allows doing stuff that normal CC modem + ME Interface -or- ME Level Emitters, etc. would allow doing, but in one block and with less lag (I hope).
Yes, you have to set up the items you want to pull out of the ME system in the interfaces (which is a slight inconvenience), but I know what items I programmed in the computers to pull, so not as big of an issue.

The ME Bridge is just a very nice convenience, that is important to me so much I looked up this mod, but I do not think that it is worth that amount of energy, which is not just a lot, it is a little bit insane, considering the power generation options (only one option).

I'm writing this, because; yes, I can change the value in the config, but a) I don't like doing stuff like that, because I might as well just play creative when I will be changing anything to work as easy as I want it to and b) a lot of people that just want to play mods, might be <unhappy/repelled/confused/annoyed/sad>, when they encounter that.
And I think that the CC in Minecraft is a really good way (at least for some kids) to learn programming.

**TLDR**
I think you should rethink that energy cost and either:
a) lower the passive amount of energy taken to 5 AE/t just like the ME Interface and make the item transfers have a cost - just like other AE machines,
b) IDK what information Forge allows you to check easily, but maybe you can check what the loaded mods are allowing to do / what functionality they are providing and just lower it when only AE2 is loaded in terms of power generation mods
c) something else that solves that for casual players ...

PS: _I figured out the phrase that I wanted to use "might be <unhappy/repelled/confused/annoyed/sad>" -> "that's going to throw them away"_

**Comment by SirEndii** [comment]
Your explanation is good. I will decrease the usage of energy to a low value like 5 or similar.

**Comment by SirEndii** [comment]
Improved in 0.7.10b
